plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -1788,6 +1798,23 @@ else if (partitionUpdate.getUpdateMode() == NEW || partitionUpdate.getUpdateMode
                         .collect(toImmutableList())));
     }
 
+    private void removeNonCurrentQueryFiles(ConnectorSession session, Path partitionPath)
+    {
+        try {
+            FileSystem fileSystem = hdfsEnvironment.getFileSystem(new HdfsContext(session), partitionPath);
+            RemoteIterator<LocatedFileStatus> iterator = fileSystem.listFiles(partitionPath, false);
+            while (iterator.hasNext()) {
+                Path file = iterator.next().getPath();
+                if (!isFileCreatedByQuery(file.getName(), session.getQueryId())) {
+                    fileSystem.delete(file, false);
+                }
+            }
+        }
+        catch (IOException ex) {
+            throw new IllegalArgumentException(format("Failed to delete partition %s files during overwrite", partitionPath), ex);
```

Comment:
Should the query id be logged as well to allow the user to manually perform a cleanup when needed?

Reply:
Good point will add it.